### PR TITLE
Bugfix/node violation mac address

### LINKED ISSUE #2738: Admin audit log: Wrong MAC address when closing a violation
When you close/release a violation of a node from the admin, it doesn't write the right MAC address and instead seems to be writing the vid of the violation that has been closed:
```
{"status":200,"context":"/node/closeViolation","action":"closeViolation","user":"lmarcotte","mac":"362","happened_at":"Mon Oct 30 15:39:15 2017"}
```

### PR DESCRIPTION
**Pull request by satkunas**
# Description
Fixes MAC in Node Violation Log (logs/httpd/admin/audit.log) on closeViolation and runViolation.

# Issue
* VID (Violation ID) was being improperly displayed as the MAC in the Node Violation Audit Logs
* Fixes #2738

# Delete branch after merge
YES 

# NEWS file entries
## Bug Fixes
* Bugfix MAC in Node::closeViolation Audit Log from VID to MAC
* Bugfix MAC in Node::runViolation Audit Log from VID to MAC

### CODE REVIEW

File: html/pfappserver/lib/pfappserver/PacketFence/Controller/Node.pm [hunk 1]
```diff
@@ -514,7 +516,10 @@ sub _triggerViolation {
 sub closeViolation :Path('close') :Args(1) :AdminRole('NODES_UPDATE') {
     my ($self, $c, $id) = @_;
     my ($status, $result) = $c->model('Node')->closeViolation($id);
-    $self->audit_current_action($c, status => $status, mac => $id);
+    my @violation = violation_view($id);
+    if (@violation) {
+        $self->audit_current_action($c, status => $status, mac => $violation[0]->{mac});
```
Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Node.pm:521]: I think the action should always be audited. 
If the violation is not found the mac should be set to either undef or 'not_found'.

File: html/pfappserver/lib/pfappserver/PacketFence/Controller/Node.pm [hunk 2]
```diff
@@ -527,7 +532,10 @@ sub closeViolation :Path('close') :Args(1) :AdminRole('NODES_UPDATE') {
 sub runViolation :Path('run') :Args(1) :AdminRole('NODES_UPDATE') {
     my ($self, $c, $id) = @_;
     my ($status, $result) = $c->model('Node')->runViolation($id);
-    $self->audit_current_action($c, status => $status, mac => $id);
+    my @violation = violation_view($id);
+    if (@violation) {
+        $self->audit_current_action($c, status => $status, mac => $violation[0]->{mac});
```
Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Node.pm:537]: @julsemaan 
Should we audit the attempt or only the success?

**Comment by julsemaan** [comment]
@satkunas I think you need to accept the CLA before we merge this one

**Review by jrouzierinverse — COMMENTED** [review]
I have some questions

File: html/pfappserver/lib/pfappserver/PacketFence/Controller/Node.pm (hunk 2 above)
Comment (julsemaan) [reply html/pfappserver/lib/pfappserver/PacketFence/Controller/Node.pm:537]: IMO, its only an action if there was something to close but both do make sense

**Pull request merged**